### ISSUE TITLE
Mining System

### ISSUE THREAD

**Issue by PatchRequest** [EveronLife/EveronLife#2]
**Summarize your proposal**
Every Life mod has a Mining System, therefore we need one too

**Describe the solution you'd like**

The goal is to have an interaction option that is generically defined.
It should take the "result" resource as a parameter, so it can be reused for every kind of farming by just changing the reward. 
The size of the reward and the time to farm should be parameters too.

**Describe alternatives you've considered**
Required Items and or Skills can be implemented at a later stage

**Comment by Bananenn** [comment]
I suggest that it's made slightly more interactive, For example, Altis life you just go to a location and action key. I think a better solution for reforger would be to go to a location and farm it from an actual rock.  Something like this.
![4f65b329-a9ea-417f-8123-c7f120021eb6](https://user-images.githubusercontent.com/75018698/171048300-a2d0f54a-b90f-4dc8-af8e-4b97af6c898e.gif)

The only problem ive ran into when trying to implement is how to delay a user from doing the same action again (want to force them to atleast somewhat more around to different "nodes")

**Comment by Arkensor** [comment]
@thebonbon yes exactly. These fall under the advanced mining category, so you should need a vehicle to be able to transport more than 2-5 ores.  The weight and volume would indeed be the way to go about this. Volume ensures that you can carry it in anything but a backpack on the player. Remember you need to be able to carry a tiny bit while mining so you can put it in the vehicle that might be parked a few meters away :)

**Comment by MarvinVomberg** [comment]
Also take a look at #92 as @Kanf has already made ores

**Comment by LuxPoseidon** [comment]
> I suggest that it's made slightly more interactive, For example, Altis life you just go to a location and action key. I think a better solution for reforger would be to go to a location and farm it from an actual rock. Something like this. ![4f65b329-a9ea-417f-8123-c7f120021eb6](https://user-images.githubusercontent.com/75018698/171048300-a2d0f54a-b90f-4dc8-af8e-4b97af6c898e.gif) [ ](https://user-images.githubusercontent.com/75018698/171048300-a2d0f54a-b90f-4dc8-af8e-4b97af6c898e.gif) ![4f65b329-a9ea-417f-8123-c7f120021eb6](https://user-images.githubusercontent.com/75018698/171048300-a2d0f54a-b90f-4dc8-af8e-4b97af6c898e.gif) [ ](https://user-images.githubusercontent.com/75018698/171048300-a2d0f54a-b90f-4dc8-af8e-4b97af6c898e.gif)
> 
> The only problem ive ran into when trying to implement is how to delay a user from doing the same action again (want to force them to atleast somewhat more around to different "nodes")

Possibly you could remove(hide) the prefab after its used for a set amount of time.

**Comment by KasperOnPC** [comment]
Just tagging in here that @kanf is planning on making the pickaxe model & texture as well. Animations are still required though.
